daft/execution/execution_step.py

```diff
@@ -630,6 +630,30 @@ def run_partial_metadata(self, input_metadatas: list[PartialPartitionMetadata])
         ]
 
 
+@dataclass(frozen=True)
+class Unpivot(SingleOutputInstruction):
+    ids: ExpressionsProjection
+    values: ExpressionsProjection
+    variable_name: str
+    value_name: str
+
+    def run(self, inputs: list[MicroPartition]) -> list[MicroPartition]:
+        return self._unpivot(inputs)
+
+    def _unpivot(self, inputs: list[MicroPartition]) -> list[MicroPartition]:
+        [input] = inputs
+        return [input.unpivot(self.ids, self.values, self.variable_name, self.value_name)]
+
+    def run_partial_metadata(self, input_metadatas: list[PartialPartitionMetadata]) -> list[PartialPartitionMetadata]:
+        # Can't derive anything.
+        return [
+            PartialPartitionMetadata(
+                num_rows=None,
+                size_bytes=None,
```

Comment:
Correct me if i'm wrong here, but can you derive num_rows from `self.len() * values.len()` ? And also size_bytes should be the same as input right?

Reply:
Oh yeah we can totally get the number of rows here. However the size_bytes is not going to be exactly the same as the values in the id columns are going to be repeated